File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun middleNode(head: ListNode?): ListNode? {
        var p: ListNode? = head
        var q: ListNode? = head
        while (p != null && p?.next != null){
            p = p?.next?.next
            q = q?.next
        }
        return q
    }
}

File: 876-middle-of-the-linked-list/876-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def middleNode(self, head):
        
        q = head
        p = head
        
        while p and p.next:
            p = p.next.next
            q = q.next
            
        return q
        